public/app/features/dashboard-scene/scene/DashboardScene.test.tsx

```diff
@@ -151,6 +158,44 @@ describe('DashboardScene', () => {
       expect(eventHandler).toHaveBeenCalledTimes(1);
     });
   });
+
+  describe('When a dashboard is restored', () => {
+    let scene: DashboardScene;
+
+    beforeEach(async () => {
+      scene = buildTestScene();
+      scene.onEnterEditMode();
+    });
+
+    it('should restore the dashboard to the selected version and exit edit mode', async () => {
+      const newVersion = 3;
+
+      const mockScene = new DashboardScene({
+        title: 'new name',
+        uid: 'dash-1',
+        version: 4,
+      });
+
+      jest.mocked(historySrv.restoreDashboard).mockResolvedValue({ version: newVersion });
+      jest.mocked(transformSaveModelToScene).mockReturnValue(mockScene);
+
+      return scene.onRestore(getVersionMock()).then((res) => {
+        expect(res).toBe(true);
+
+        expect(scene.state.version).toBe(newVersion);
+        expect(scene.state.title).toBe('new name');
+        expect(scene.state.isEditing).toBe(false);
+      });
+    });
+
+    it('should return early if historySrv does not return a valid version number', async () => {
+      jest.mocked(historySrv.restoreDashboard).mockResolvedValue({ version: null });
```

Comment:
Have you tried with `Infinity` or `NaN`? we're using [`isNumber`](https://lodash.com/docs/4.17.15#isNumber) checker to know if it is a valid number. Probably using `Number.isInteger()` would be safer. WDYT?

Reply:
I agree with you, `isInteger` is better suited here. Modified the code and made the test more explicit. Thank you!